CODE_GENERATION.md

```diff
@@ -0,0 +1,290 @@
+# Contributing New Resource Using ACK
+
+Provider AWS uses ACK code generation pipeline to generate the boilerplate for
+CRDs and their controllers. Most of the generated code is ready to go but there
+are places where we need Crossplane-specific functionality or some hacks to work
+around non-generic behavior of the AWS API.
+
+This guide shows how to get a new resource support up and running step by step.
+
+## Code generation
+
+This is the very first step. You need to clone [ACK repository](https://github.com/aws/aws-controllers-k8s/) and run the following
+command in the root directory:
+
+```console
+go run -tags codegen cmd/ack-generate/main.go crossplane <ServiceID> --provider-dir <provider-aws directory>
+```
+
+`ServiceID` list is found here: https://github.com/aws/aws-sdk-go/tree/master/models/apis
+For example, in order to generate Lambda resources, you need to use `lambda` as
+Service ID.
+
+The first run will take a while since it clones AWS SDK. Once it is completed,
+you'll see new folders in `apis` and `pkg/controller` directories.
+By default, it generates every resource it can recognize but that's not what you
+want if you'd like to add support for only a few resource. In order to ignore some
+of the generated resources, you need to create a `apis/<serviceid>/v1alpha1/generator-config.yaml`
+file with the following content:
+```yaml
+ignore:
+  resource_names:
+    - <CRD Name you want to ignore>
+```
+
+When you re-run the generation with this configuration, existing files won't be
+deleted. So you may want to delete everything in `apis/<serviceid>/v1alpha1` except
+`generator-config.yaml` and then re-run the command.
+
+If the generation fails for any reason, please open an issue.
```

Comment:
Should we open issues against ACK, or Crossplane?

Reply:
It should be ACK since the code is there. Added some clarification.